src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -95,6 +116,60 @@ public static Email parseEmail(String email) throws ParseException {
         return new Email(trimmedEmail);
     }
 
+    /**
+     *  Parses a {@code String company} into a {@code Company}.
+     *  Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code company} is invalid.
+     */
+    public static JobCompany parseCompany(String company) throws ParseException {
+        requireNonNull(company);
+        String trimmedCompany = company.trim();
+        if (!JobCompany.isValidCompany(trimmedCompany)) {
+            throw new ParseException(JobCompany.MESSAGE_CONSTRAINTS);
+        }
+        return new JobCompany(trimmedCompany);
+    }
+
+    /**
+     *  Parses a {@code String salary} into a {@code JobSalary}.
+     *  Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code salary} is invalid.
+     */
+    public static JobSalary parseSalary(String salary) throws ParseException {
+        requireNonNull(salary);
+        String trimmedSalary = salary.trim();
+        if (!JobSalary.isValidSalary(trimmedSalary)) {
+            throw new ParseException(JobSalary.MESSAGE_CONSTRAINTS);
+        }
+        return new JobSalary(trimmedSalary);
+    }
+
+    /**
+     *  Parses a {@code String requirements} into a {@code JobRequirements}.
+     *  Leading and trailing whitespaces will be trimmed.
+     *  Note that there is no validation for this.
+     *  todo: change requirements to tags for the matching feature
+     */
+    public static JobRequirements parseRequirements(String requirements) {
+        requireNonNull(requirements);
+        String trimmedRequirements = requirements.trim();
+        return new JobRequirements(trimmedRequirements);
+    }
+
+    /**
+     *  Parses a {@code String description} into a {@code JobDescription}.
+     *  Leading and trailing whitespaces will be trimmed.
+     *  Note that there is no validation for this.
+     *
+     */
+    public static JobDescription parseDescription(String description) {
+        requireNonNull(description);
+        String trimmedDescription = description.trim();
```

Comment:
Just curious, what would the description be if it wasn't provided in the command? Is it empty string still it can't be `Null`?

Reply:
I'll be honest, I have no idea why `requireNonNull()` is utilized, as including it or not doesn't result in the app breaking (and the displayed text is the same). 

 Prior to calling the methods in `parserUtil`, usually the input goes through `ArgMultiMap`, where we then use `ArgMultiMap.getValue(PREFIX).get()`. However, `getValue(PREFIX)`, in the event a PREFIX doesn't exist, returns an `Optional.empty()` and calling `get()` on that returns an `Exception`, so perhaps it has something to do with that?

Reply:
Wait so currently we can't create a job without specifying requirements and description?

I do think having requireNonNull() should be included just to be safe, because theoretically the caller of this method can just pass in Null, causing error when trimming. 

What I wanted to know is that, since what would optional arguments be stored as in json if we do not specify them in command.

Reply:
Oh my bad, I just saw #71, I think you can resolve this now

Reply:
Oh that was what you meant XD. 

Yea, I have identified that as a bug but also at the same time, it is also because I am currently using `requirements` and `description` which might be replaced by `tags`. So I'm holding off on fixing the bug until I am clearer on what the feature spec should be.

Marking this as resolved for now.